eclass/java-pkg-simple.eclass

```diff
@@ -455,6 +556,29 @@ java-pkg-simple_src_compile() {
 	fi
 }
 
+# @FUNCTION: java-pkg-simple_src_compile
+# @DESCRIPTION:
+# src_compile for simple bare source java packages. Compiles a single
+# jar via java-pkg-simple_compile_jar and calls
+# java-pkg-simple_call_ejavadoc to generate documentation if USE FLAG
+# 'doc' exists and is set.
+java-pkg-simple_src_compile() {
+	java-pkg-simple_compile_jar
+
+	# javadoc
+	if has doc ${JAVA_PKG_IUSE} && use doc; then
+		if has binary ${JAVA_PKG_IUSE} && use binary; then
+			ewarn "Generation of documentation is not supported when installing binary (USE=binary)."
+		elif [[ ${JAVADOC_SRC_DIRS} ]]; then
+			eqawarn "JAVADOC_SRC_DIRS is set and java-pkg-simple_src_compile called."
+			eqawarn "Please use JAVADOC_SRC_DIRS together with java-pkg-simple_compile_jar/java-pkg-simple_call_ejavadoc instead."
+			eqawarn "This will become an error in the future."
```

Comment:
> I replaced `einfo` with `eqawarn` and adjusted the message. @fordfrog, @vaukai: Are you fine with the wording?

personally, i'd prefer the long version:
```
"When setting JAVADOC_SRC_DIRS use java-pkg-simple_compile_jar 
instead of java-pkg-simple_src_compile for the jar files 
and call java-pkg-simple_call_ejavadoc for javadoc."
```

Reply:
Yes, your version reads a lot better. Thanks!